src/nlsolve_powerflow.jl

```diff
@@ -64,84 +82,84 @@ function _dogleg!(p::Vector{Float64},
 end
 
 function _trust_region_step(time_step::Int,
-    cache::NLCache,
+    X::StateVectorCache,
     linSolveCache::KLULinSolveCache{Int32},
-    pf::PolarPowerFlow,
+    R::ACPowerFlowResidual,
     J::ACPowerFlowJacobian,
     delta::Float64,
     eta::Float64 = DEFAULT_TRUST_REGION_ETA)
     numeric_refactor!(linSolveCache, J.Jv)
 
     # find proposed next point.
-    _dogleg!(cache.p, cache.p_c, cache.pi, cache.r, linSolveCache, J.Jv, delta)
-    cache.x .+= cache.p
+    _dogleg!(X.p, X.p_c, X.pi, X.r, linSolveCache, J.Jv, delta)
+    X.x .+= X.p
 
     # use cache.pi as temporary buffer to store old residual
     # to avoid recomputing if we don't change x.
-    oldResidual = cache.pi
-    copyto!(oldResidual, pf.residual)
-    pf(cache.x)
+    oldResidual = X.pi
+    copyto!(oldResidual, R.Rv)
+    R(X.x, time_step)
 
     # Ratio of actual to predicted reduction
-    LinearAlgebra.mul!(cache.r_predict, J.Jv, cache.p)
-    cache.r_predict .+= cache.r
+    LinearAlgebra.mul!(X.r_predict, J.Jv, X.p)
+    X.r_predict .+= X.r
     rho =
-        (sum(abs2, cache.r) - sum(abs2, pf.residual)) /
-        (sum(abs2, cache.r) - sum(abs2, cache.r_predict))
+        (sum(abs2, X.r) - sum(abs2, R.Rv)) /
+        (sum(abs2, X.r) - sum(abs2, X.r_predict))
     if rho > eta
         # Successful iteration
-        cache.r .= pf.residual
+        X.r .= R.Rv
```

Comment:
I don't like to have single letter variables. Here I can't know what `X` is. In the developers guidelines for Sienna we discourage this.